src/base/kaldi-math.h

```diff
@@ -69,9 +70,9 @@
 #endif
 
 
-#define KALDI_ISNAN std::isnan
-#define KALDI_ISINF std::isinf
-#define KALDI_ISFINITE(x) std::isfinite(x)
+#define KALDI_ISNAN(x) _isnan(x)
+#define KALDI_ISINF(x) (!_finite(x))
+#define KALDI_ISFINITE(x) _finite(x)
 
```

Comment:
I think _isnan() and _finite() are Windows SDK only, i.e. I  don't think this compile on Linux anymore 
are you sure std::isnan and std::isinf and std::isfinite are not on Windows? I was able to compile the solution a couple of months back... Perhaps your VS is old(er)?